client/src/components/CountrySelector/index.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { COUNTRIES } from '../../constants/countries';
+
+import './styles.css';
+
+class CountrySelector extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const defaultCountry = this.getCountry(props);
+
+    this.state = {
+      displayedCountries: COUNTRIES,
+      isListVisible: false,
+      currentCountry: Object.keys(defaultCountry).length
+        ? defaultCountry
+        : COUNTRIES[0],
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const newCountry = this.getCountry(nextProps);
+    const { currentCountry } = this.state;
+
+    if (newCountry.countryName !== currentCountry.countryName) {
+      this.setState({ currentCountry: newCountry });
+    }
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    const { currentCountry } = this.state;
+    const { currentCountry: newCountry } = nextState;
+
+    if (currentCountry.countryName !== newCountry.countryName) {
+      this.props.getSelectedCountry(newCountry);
+    }
+  }
+
+  getCountry = (props) => {
+    let country = {};
+
+    if (props.defaultCountry) {
```

Comment:
You should use `this.props.defaultCountry`.

Reply:
No, I do not agree...
I have implemented this function with param **props** with an intention to be able to use it for `this.props` as well as for `nextProps` from **componentWillReceiveProps**. So, in this case, I need to use `props.defaultCountry`because it depends with which param I call the function `getCountry`.